Classes/Issues/GithubClient+Issues.swift

```diff
@@ -389,4 +389,87 @@ extension GithubClient {
         })
     }
 
+    func addAssignees(
+        previous: IssueResult,
+        owner: String,
+        repo: String,
+        number: Int,
+        assignees: [IssueAssigneeViewModel]
+        ) {
+        guard let actor = userSession?.username else { return }
+
+        let oldAssigness = Set<String>(previous.assignee.users.map { $0.login })
+        let newAssignees = Set<String>(assignees.map { $0.login })
+
+        var newEvents = [IssueRequestModel]()
+        var added = [String]()
+        var removed = [String]()
+
+        for old in oldAssigness {
+            if !newAssignees.contains(old) {
+                removed.append(old)
+                newEvents.append(IssueRequestModel(
+                    id: UUID().uuidString,
+                    actor: actor,
+                    user: old,
+                    date: Date(),
+                    event: .unassigned,
+                    width: 0 // will be inflated when asked
+                    ))
+            }
+        }
+        for new in newAssignees {
+            if !oldAssigness.contains(new) {
+                added.append(new)
+                newEvents.append(IssueRequestModel(
+                    id: UUID().uuidString,
+                    actor: actor,
+                    user: new,
+                    date: Date(),
+                    event: .assigned,
+                    width: 0 // will be inflated when asked
+                ))
+            }
+        }
+
+        let optimistic = previous.updated(
+            assignee: IssueAssigneesModel(users: assignees, type: .assigned),
+            timelinePages: previous.timelinePages(appending: newEvents)
+        )
+
+        let cache = self.cache
+        cache.set(value: optimistic)
+
+        let path = "repos/\(owner)/\(repo)/issues/\(number)/assignees"
+
+        let handler: (Int, Int?) -> Void = { (expect, status) in
+            if status != expect {
+                cache.set(value: previous)
+                ToastManager.showGenericError()
+            }
+        }
+
+        // https://developer.github.com/v3/issues/assignees/#add-assignees-to-an-issue
+        if added.count > 0 {
+            request(GithubClient.Request(
+                path: path,
+                method: .post,
+                parameters: ["assignees": added]
+            ) { (response, _) in
+                handler(201, response.response?.statusCode)
+            })
+        }
+
+        // https://developer.github.com/v3/issues/assignees/#remove-assignees-from-an-issue
```

Comment:
These endpoints are lame. Unlike labelling, adding and removing are in **separate** endpoints. There's definitely the case where one of these fails and the other succeeds, then you reset the data back to the _original_ state (not including success or failure), but I don't really care at this point. Should be rare.

Reply:
That is such a stupid limitation, what...